sdg/DocumentationService.py

```diff
@@ -0,0 +1,159 @@
+import os
+import sdg
+import numpy
+from slugify import slugify
+
+class DocumentationService:
+    """HTML generation to document outputs built with this library.
+
+    Note that this is meant to document particular builds, not the library in
+    general. The idea is that each time this library is used to build/convert
+    some SDG-related data, this class can be used to generate human-friendly
+    HTML pages documenting the specifics of the build (such as endpoint URLs).
+    """
+
+
+    def __init__(self, outputs, folder='_site', branding='Build docs', languages=None, intro=''):
+        """Constructor for the DocumentationService class.
+
+        Parameters
+        ----------
+        outputs : list
+            List of objects inheriting from OutputBase
+        """
+        self.outputs = outputs
+        self.folder = folder
+        self.branding = branding
+        self.intro = intro
+        self.slugs = []
+        self.languages = [] if languages is None else languages
+
+
+    def generate_documentation(self):
+        """Generate the HTML pages for documentation of all of the outputs."""
+        pages = []
+        for output in self.outputs:
+            title = output.get_documentation_title()
+            content = output.get_documentation_content(self.languages)
+            pages.append({
+                'title': title,
+                'filename': self.create_filename(title),
+                'content': content
+            })
+
+        if not os.path.exists(self.folder):
+            os.makedirs(self.folder)
```

Comment:
`os.makedirs` has a [built-in check](https://docs.python.org/3/library/os.html#os.makedirs) for checking if the folder already exists, if you'd like to do without the `if` statement (`exist_ok=True`)

Reply:
Cool, just made that change.